Sources/MuxStatsGoogleIMAPlugin/MUXSDKIMAAdsListener.m

```diff
@@ -0,0 +1,317 @@
+//
+//  MUXSDKIMAAdsListener.m
+//  MuxStatsGoogleIMAPlugin
+//
+
+#import <Foundation/Foundation.h>
+#import "MUXSDKIMAAdsListener.h"
+
+@interface MUXSDKIMAAdsListener ()
+
+@property (nonatomic, nonnull) MUXSDKPlayerBinding *playerBinding;
+
+@property (assign) BOOL sendAdplayOnStarted;
+@property (assign) BOOL isPictureInPicture;
+@property (assign) BOOL usesServerSideAdInsertion;
+@property (assign) BOOL adRequestReported;
+@property (assign) NSString *adTagURL;
+
+@end
+
+@implementation MUXSDKIMAAdsListener
+
+
+- (instancetype)initWithPlayerBinding:(MUXSDKPlayerBinding *)binding
+      monitoringAdsLoader:(nullable IMAAdsLoader *)adsLoader {
+    return [self initWithPlayerBinding:binding
+                               options:MUXSDKIMAAdsListenerOptionsNone
+                   monitoringAdsLoader:adsLoader];
+}
+
+- (instancetype)initWithPlayerBinding:(MUXSDKPlayerBinding *)binding
+                    options:(MUXSDKIMAAdsListenerOptions)options
+        monitoringAdsLoader:(IMAAdsLoader *)adsLoader {
+    self = [super init];
+
+    if (self) {
+        self.customerAdsLoaderDelegate = adsLoader.delegate;
```

Comment:
```suggestion
        _customerAdsLoaderDelegate = adsLoader.delegate;
```

Reply:
In Objective-C we want to avoid accessing self while initializing.

To set initial state we access properties underlying instance variables. By convention these are automatically generated by the compiler. Their naming follows the name of the property the back, prefixed with `_`.